pkg/controller/pxc/users.go

```diff
@@ -326,3 +327,41 @@ func sysUsersSecretDataChanged(newHash string, usersSecret *corev1.Secret) (bool
 func sha256Hash(data []byte) string {
 	return fmt.Sprintf("%x", sha256.Sum256(data))
 }
+
+func (r *ReconcilePerconaXtraDBCluster) reconcileSyncPXCUsersWithProxySQL(cr *api.PerconaXtraDBCluster) {
+	if cr.Status.Status != api.AppStateReady {
+		return
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	stopCh := make(chan error, 1)
+
+	go func() {
+		r.syncUsersChan <- 1
+		defer func() {
+			if len(r.syncUsersChan) > 0 {
+				<-r.syncUsersChan
+				log.Info("done")
+			}
+		}()
+
+		stopCh <- r.syncPXCUsersWithProxySQL(cr)
+	}()
+	for {
+		select {
+		case <-ctx.Done():
+			cancel()
```

Comment:
Will be canceled in defer (line 337)

Reply:
https://github.com/percona/percona-xtradb-cluster-operator/pull/433/commits/4f1fcc05ab1189433fb026af59b58d8abf1bfb28